src/librustc_mir/borrow_check/prefixes.rs

```diff
@@ -26,8 +26,9 @@ impl<'tcx> IsPrefixOf<'tcx> for Place<'tcx> {
             }
 
             match *cursor {
-                Place::Promoted(_) |
-                Place::Local(_) | Place::Static(_) => return false,
+                Place::Base(PlaceBase::Promoted(_)) |
+                Place::Base(PlaceBase::Local(_)) |
+                Place::Base(PlaceBase::Static(_)) => return false,
```

Comment:
Just use a single pattern (`Place::Base(_)`)

Reply:
Hmm kind of disagree with this one. What if we add something in `Base` that needs a different treatment. For instance the All Hands docs talks about adding `Deref` here. Unsure if that will need to be treated differently.

Reply:
Sure, fully listing all variants is a good regression check.